### PR TITLE
Replaced abort() by Flask Exceptions, closes #3

### LINKED ISSUE #3: Use Flask Exceptions
We should use [Flask Exceptions](http://flask.pocoo.org/docs/0.11/patterns/apierrors/) instead of `abort()` to signal errors for invalid API usage.

### PR DESCRIPTION
**Pull request by AlissaWe**
- new exception classes
- new error handlers
- replaced every use of `abort()` with `raise exception()`

### CODE REVIEW

File: ocyco/api/decorators.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from werkzeug.exceptions import abort
 
 from ocyco.models.users import Users
+from ocyco.api.exceptions import OcycoException, ParameterInvalidException, ParameterMissingException, NotFoundException, ConflictExistingObjectException, MultipleMatchesException, PhotonException
```
Comment (rleh) [inline ocyco/api/decorators.py]: You don't need to import all exception, only `MultipleMatchesException` is needed here.

File: ocyco/api/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        rv = dict(self.payload or())
+        rv['message'] = self.message
+        return rv
+
+
+class ParameterInvalidException(OcycoException):
+    status_code = 400
+
+    def __init__(self, message, status_code=None, payload=None):
+        Exception.__init__(self)
```
Comment (rleh) [inline ocyco/api/exceptions.py]: Does this work? I think you should call `OcycoException.__init__(self)` here...

File: ocyco/api/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        if status_code is not None:
+            self.status_code = status_code
+        self.payload = payload
+
+
+class ParameterMissingException(OcycoException):
+    status_code = 400
+
+    def __init__(self, message, status_code=None, payload=None):
+        Exception.__init__(self)
```
Comment (rleh) [inline ocyco/api/exceptions.py]: see above

File: ocyco/api/geo.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 from werkzeug.exceptions import abort
 from ocyco.database import db
+from ocyco.api.exceptions import OcycoException, ParameterInvalidException, ParameterMissingException, NotFoundException, ConflictExistingObjectException, MultipleMatchesException, PhotonException
```
Comment (rleh) [inline ocyco/api/geo.py]: Only `ParameterMissingException` is needed, see above.

File: ocyco/api/tracks.py [hunk 5]
```diff
@@ -10,6 +10,7 @@
 from ocyco.api.decorators import requires_authentication
 from ocyco.models.tracks import Tracks, TrackPoints
 from ocyco.utils import get_city_by_coordinates
+from ocyco.api.exceptions import OcycoException, ParameterInvalidException, ParameterMissingException, NotFoundException, ConflictExistingObjectException, MultipleMatchesException, PhotonException
```
Comment (rleh) [inline ocyco/api/tracks.py]: see above

**Review by rleh — CHANGES_REQUESTED** [review]

**Review by rleh — APPROVED** [review]

**Comment by Harry-R** [comment]
Thanks for your contribution! 👍

**Pull request merged**